pulsar-metadata/src/main/java/org/apache/pulsar/metadata/coordination/impl/LockManagerImpl.java

```diff
@@ -115,9 +115,11 @@ private void handleSessionEvent(SessionEvent se) {
 
     private void handleDataNotification(Notification n) {
         if (n.getType() == NotificationType.Deleted) {
-            locks.stream()
-                    .filter(l -> l.getPath().equals(n.getPath()))
-                    .forEach(l -> l.lockWasInvalidated());
+            for (ResourceLockImpl<T> lock : locks) {
```

Comment:
@linlinnn is right, this code is only rewriting the loop without the stream.

probably the best way is to scan the collection and create a list of locks to be processed, then you scan the list

```
List<ResourceLockImpl<T>> invalidated = locks.stream()
                    .filter(l -> l.getPath().equals(n.getPath())).collect(Collectors.toList());

invalidated.forEach(l -> l.lockWasInvalidated())
                    
```

Reply:
@eolivelli @linlinnn  Thanks for your reply, i update the code, PTAL.

Reply:
I think we cannot fix it in this way.
Because if other thread removes the element during create the list, it will throw `ConcurrentModificationException` when call `forEachRemaining(action)`.
The simpliest way is to use `CopyOnWriteArraySet` instead of `HashSet`,
or add a lock.

Reply:
@linlinnn  Thanks for your feedback. use `CopyOnWriteArraySet` instead of `HashSet` is the simpliest way.